src/browser/preload_scripts.ts

```diff
@@ -80,8 +80,8 @@ function downloadScript(identity: Identity, preloadScript: PreloadScript): Promi
 
 export async function loadScripts(identity: Identity): Promise<PreloadScriptWithContent[]|any> {
     const options = Window.getOptions(identity);
-    const { preloadScripts } = convertToElectron(options);
-    const promises = preloadScripts.map((preloadScript: PreloadScript) => loadScript(identity, preloadScript));
+    const { preloadScripts = [] }: {preloadScripts: Array<PreloadScript>} = convertToElectron(options);
```

Comment:
Shouldn't this be `const { preloadScripts = [] }: PreloadScript[] = convertToElectron(options);`?

Reply:
I actually prefer Array<T>

Reply:
But I think the type is wrong though or am I missing something?
`const { preloadScripts = [] }: { preloadScripts: Array<PreloadScript> } = convertToElectron(options);`
I think it should be:
`const { preloadScripts = [] }: Array<PreloadScript> = convertToElectron(options);`
as resulting `preloadScripts` is an array, not an object with a property...

Reply:
ahh, that is the correct syntax for destructing.

Reply:
meaning the code there is correct, preloadScripts will be an array of PreloadScript.

Reply:
https://www.typescriptlang.org/docs/handbook/variable-declarations.html#destructuring